core/domain/suggestion_services_test.py

```diff
@@ -2084,6 +2094,1155 @@ def test_create_translation_contribution_stats_from_model(self) -> None:
+            feconf.SUGGESTION_TYPE_ADD_QUESTION,
+            feconf.ENTITY_TYPE_SKILL, skill_id, 1,
+            self.author_id, suggestion_change, 'test description')
+
+    def _create_skill(self) -> str:
+        """Creates a topic for a question.
+
+        Returns:
+            str. A topic ID.
+        """
+        skill_id = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            skill_id, self.author_id, description='description')
+        return skill_id
+
+    def _create_topic(self, first_skill_id: str, second_skill_id: str) -> str:
+        """Creates a topic for a question.
+
+        Args:
+            first_skill_id: str. ID of the first skill.
+            second_skill_id: str. ID of the second skill.
+
+        Returns:
+            str. A topic ID.
+        """
+        topic_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id, 'topic_admin', name='Topic1',
+            abbreviated_name='topic-three', url_fragment='topic-three',
+            description='Description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[first_skill_id, second_skill_id],
+            subtopics=[], next_subtopic_id=1)
+        return topic_id
+
+    def test_update_question_contribution_stats_when_submitting(self) -> None:
+        # Steps required in the setup phase before testing.
+        # 1. Save new skills.
+        # 2. Save a topic assigning skills for it.
+        # 4. Create a question suggestion.
+        skill_id_1 = self._create_skill()
+        skill_id_2 = self._create_skill()
+        topic_id = self._create_topic(skill_id_1, skill_id_2)
+        initial_suggestion = self._create_question_suggestion(skill_id_1)
+        latest_suggestion = self._create_question_suggestion(skill_id_2)
+        suggestion_services.update_question_contribution_stats_at_submission(
+            initial_suggestion
+        )
+        suggestion_services.update_question_contribution_stats_at_submission(
+            latest_suggestion
+        )
+
+        question_contribution_stats_model = (
+            suggestion_models.QuestionContributionStatsModel.get(
+                self.author_id, topic_id
+            )
+        )
+        # Assert question contribution stats before the review.
+        # At this point we can confirm that there should be an associated
+        # question contribution stat object for the given IDs since we have
+        # called update_question_contribution_stats_at_submission function to
+        # create/update question contribution stats.
+        assert question_contribution_stats_model is not None
+        self.assertEqual(
+            question_contribution_stats_model.submitted_questions_count,
+            2
+        )
+        self.assertEqual(
+            question_contribution_stats_model.accepted_questions_count,
+            0
+        )
+
+        suggestion_services.accept_suggestion(
```

Comment:
Aren't the below lines in this test redundant with the below test that tests the accept behavior?

Reply:
Yeah removed